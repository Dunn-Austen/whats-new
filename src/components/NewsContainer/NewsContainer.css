* {
  background-color: #e8d3dc;
}
