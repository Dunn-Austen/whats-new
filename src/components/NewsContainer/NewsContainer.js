import React from 'react';
import NewsArticle from '../NewsArticle/NewsArticle';
import './NewsContainer.css'

const NewsContainer = (props) => {

  const newsArticles = props.localNews.map(newsBlock => {
    return (
      <NewsArticle
        headline={newsBlock.headline}
        img={newsBlock.img}
        description={newsBlock.description}
        url={newsBlock.url}
        id={newsBlock.id}
        key={newsBlock.id}
      />
    )
  });

  return (
    <div className='news-container'>
      {newsArticles}
    </div>
  )
};

export default NewsContainer;
