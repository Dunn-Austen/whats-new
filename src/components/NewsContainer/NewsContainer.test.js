import React from 'react';
import { shallow } from 'enzyme';
import NewsContainer from './NewsContainer';

describe('NewsContainer', () => {

  it('should match the snapshot with all data correctly passed', () => {
    const wrapper = shallow(<NewsContainer
      genreDisplayed={[{id: 1, headline: 'A' , img: 'B' , description: 'C', url: 'D'}]}
      />);

    expect(wrapper).toMatchSnapshot();
  })
})
