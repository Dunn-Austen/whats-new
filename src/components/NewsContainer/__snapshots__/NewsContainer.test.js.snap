// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsContainer should match the snapshot with all data correctly passed 1`] = `
<section
  className="news-container"
>
  <NewsArticle
    description="C"
    headline="A"
    id={1}
    img="B"
    key="1"
    url="D"
  />
</section>
`;
